### PR TITLE
Added basic integration for the cancel feature.

### PR DESCRIPTION
**Pull request by skazi0**
This is useful for testing the begin upgrade without a need for restoring previous cloud state.

### CODE REVIEW

**Review by Itxaka — APPROVED** [review]
Looking good!

File: assets/app/widgets/crowbar-upgrade-nav/crowbar-upgrade-nav.directive.js [hunk 1]
```diff
@@ -11,7 +11,23 @@
...
-            }
+            },
+            controller: CrowbarUpgradeNavController,
+            controllerAs: 'upgradeNavVm'
         };
     }
+
+    CrowbarUpgradeNavController.$inject = ['$state', 'upgradeFactory'];
+
+    function CrowbarUpgradeNavController($state, upgradeFactory) {
```
Comment (santiph) [inline assets/app/widgets/crowbar-upgrade-nav/crowbar-upgrade-nav.directive.js]: Widgets, as they've been conceived, are meant to be reusable components across different projects in the future simply by adding the module as a dependency. Therefore, they should not depend on other modules, like `suseData` nor on the features modules to determine the states and consume the API.

Please, provide those attributes through the controller and their scopes.

**Review by santiph — CHANGES_REQUESTED** [review]

**Review by Itxaka — APPROVED** [review]
Still looking good to me

**Comment by skazi0** [comment]
rebased

File: assets/app/widgets/crowbar-upgrade-nav/crowbar-upgrade-nav.directive.jade [hunk 2]
```diff
@@ -1,4 +1,4 @@
-button.btn.btn-default.pull-left(type='button')
+button.btn.btn-default.pull-left(type='button', ng-click="onCancel()")
   span(aria-hidden="true")
   | Cancel Upgrade
```
Comment (santiph) [inline assets/app/widgets/crowbar-upgrade-nav/crowbar-upgrade-nav.directive.jade]: translation needed

File: assets/app/features/upgrade/upgrade.controller.js [hunk 3]
```diff
@@ -47,5 +49,14 @@
         function isLastStep() {
             return vm.steps.list[vm.steps.list.length - 1] === upgradeStepsFactory.activeStep;
         }
+
+        /**
+         * Trigger cancellation of the upgrade process and go back to landing page
+         */
+        function cancelUpgrade() {
+            upgradeFactory.cancelUpgrade().then(function(/* response */) {
+                $state.go('upgrade-landing');
```
Comment (santiph) [inline assets/app/features/upgrade/upgrade.controller.js:60]: After cancelation, user needs to be redirected to crowbar dashboard.
Completely out of the scope of this PR, but I'd like to ask you to add a comment in the code and https://trello.com/c/aP3CKeqb/28-2-17-cancellation-routine-page-creation to not lose track of it

File: assets/app/data/crowbar/services/upgrade.factory.spec.js [hunk 4]
```diff
@@ -153,6 +154,32 @@ describe('Upgrade Factory', function () {
...
+                expect(prepareNodesPromise['success']).toEqual(jasmine.any(Function));
+            });
+
+            it('when resolved, it returns the cancelUpgrade response', function () {
+                cancelUpgradePromise.then(function (response) {
+                    expect(response.status).toEqual(200);
+                });
+                $httpBackend.flush();
+            });
+        });
```
Comment (santiph) [inline assets/app/data/crowbar/services/upgrade.factory.spec.js:253]: When executed, it's supposed to also redirect the user to crowbar dashboard... Again, out of the scope, but I'd appreciate a comment in the code and the trelo card: https://trello.com/c/aP3CKeqb/28-2-17-cancellation-routine-page-creation

**Review by santiph — CHANGES_REQUESTED** [review]
Good implementation!
Please address the minor request so we don't lose track of them and consider this approved.

File: assets/app/data/crowbar/services/upgrade.factory.spec.js (hunk 4 above)
Comment (skazi0) [reply assets/app/data/crowbar/services/upgrade.factory.spec.js:253]: The redirect should be done in the controller so the service doesn't need any changes.

File: assets/app/features/upgrade/upgrade.controller.js (hunk 3 above)
Comment (skazi0) [reply assets/app/features/upgrade/upgrade.controller.js:60]: Comment added in code. I don't have write access to this trello so can't do anything there.

**Comment by Itxaka** [comment]
[(3) S51P10 (S50P9 S49P20) #6 Layout - Navigation bar](https://trello.com/c/LNr710jO/104-3-s51p10-s50p9-s49p20-6-layout-navigation-bar)

**Review by santiph — APPROVED** [review]

**Comment by santiph** [comment]
Approved as a temporary solution to be properly implemented (or completely removed, depending on the PO decision) in a future sprint

**Pull request merged**